### ISSUE TITLE
Add area_set_monitorable() to Physics2DServer

### ISSUE THREAD

**Issue by CherryPetrichor** [godotengine/godot-proposals#2244]
### Describe the project you are working on
Bullet Hell
### Describe the problem or limitation you are having in your project
Bullets are created using Area2D's with Physics2DServer. By default Monitoring is on, which causes massive lag. There is no way to turn of Monitoring through Physics2DServer.
### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Adding area_set_monitoring() to Physics2DServer. In my game only the object the bullets will collide with need to be monitoring, the bullets just need to be monitorable.
### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
I don't know engine specifics, but this should not be hard to implement considering it's just toggling an area2d variable.
### If this enhancement will not be used often, can it be worked around with a few lines of script?
There is no workaround. RIDs are opaque, and all that is given by Physics2DServer.
### Is there a reason why this should be core and not an add-on in the asset library?
Area_set_monitorable() already exists, and this is very low-level.

**Comment by Xrayez** [comment]
> Bullets are created using `Area2D`'s with Physics2DServer. By default Monitoring is on, which causes massive lag. There is no way to turn of Monitoring through Physics2DServer.

This probably only applies for `Area[2D]` nodes which enables this behavior by default (frontend), but not physics server (backend).

According to https://github.com/godotengine/godot/blob/master/scene/2d/area_2d.cpp#L362-L379, it seems to be possible to do this already, otherwise it wouldn't be possible to have `Area2D` functionality in core as of now. and `area_set_monitor_callback` and `area_set_area_monitor_callback` are exposed to script currently.

**Comment by Tootamcboota** [comment]
Is your implication that I can set Monitoring with area_set_area_monitor_callback?

**Comment by Xrayez** [comment]
Yes, I haven't used those myself but that seems like it's done internally. In any case, you'll need to compare the performance with using `Area2D` nodes and using `Physics2DServer`, because I think that "monitoring" may be actually disabled by default.

**Comment by pouleyKetchoupp** [comment]
You can try to do something like this from gdscript:
```
# Disable body detection.
Physics2DServer.area_set_monitor_callback(area_id, null, "")
# Disable area detection.
Physics2DServer.area_set_area_monitor_callback(area_id, null, "")
```

But yeah, I would also expect that creating areas through the physics server wouldn't have these enabled by default, so maybe the performance problems come from something else.

Could you open an issue with a minimum project to test your use case? The best is to start with confirming where the lag comes from, then it will be easier to decide how to fix it.
